### ISSUE TITLE
__asan_option_detect_stack_use_after_return on docker ci build since 0.4.1

### ISSUE THREAD

**Issue by doits** [brianmario/mysql2#685]
I'm trying to use version 0.4.1 in my docker ci instance (ubuntu:vivid image). It builds fine but doesn't work because it ends with this error:

```
/cache/gems/stellenticket/ubuntu-ci_latest/ruby/2.1.0/gems/activesupport-4.2.4/lib/active_support/dependencies.rb:274:in `require': /cache/gems/stellenticket/ubuntu-ci_latest/ruby/2.1.0/extensions/x86_64-linux/2.1.0/mysql2-0.4.1/mysql2/mysql2.so: undefined symbol: __asan_option_detect_stack_use_after_return - /cache/gems/stellenticket/ubuntu-ci_latest/ruby/2.1.0/extensions/x86_64-linux/2.1.0/mysql2-0.4.1/mysql2/mysql2.so (LoadError)
```

Looks like `libasan` was not linked correctly?

The error does not happen with version 0.3.20.

**Comment by sodabrew** [comment]
I noticed the same problem on the new XCode 7 compiler on Travis. I will dig into this asap.

**Comment by maxehmookau** [comment]
Any progress on this? Started having the same issue this morning too.

**Comment by sodabrew** [comment]
Sorry, I let this slip to the back-burner. I was not able to figure out how to fix the asan linkage. I'll be sure to pick it up asap with a fix to remove the extra ENV['CI'] compile flags. @tamird

**Comment by tamird** [comment]
Something doesn't make sense; `libasan` is being linked alright, but then not found at runtime? This seems like a travis issue.

**Comment by sodabrew** [comment]
Repros with XCode 7 on OS X, too.

**Comment by tamird** [comment]
Confirmed that this repros locally with XCode 7. Still digging.

**Comment by tamird** [comment]
References to similar issues elsewhere:
https://github.com/google/sanitizers/issues/380
https://github.com/JuliaLang/julia/issues/13858

**Comment by maxehmookau** [comment]
FWIW, I'm using codeship rather than travis for CI and the issue appears to be the same. I've locked the gem to 0.4.0 and it's fine for now.

**Comment by sodabrew** [comment]
Resolved by #703
